client/blocks/product-selector/index.jsx

```diff
@@ -268,7 +309,7 @@ export class ProductSelector extends Component {
 			translate,
 		} = this.props;
 
-		if ( isEmpty( storeProducts ) || fetchingSitePurchases ) {
+		if ( isEmpty( storeProducts ) || fetchingSitePurchases || fetchingSitePlans ) {
```

Comment:
I think we need to wait before both the site fetchingSitePurchases and fetchingSitePlans are done before we stop showing the placeholder for the product card. 

I noticed that in come cases there is a flickr in state. Where we show the radio toggles and then we show the you already have the backup due to the plan.

Reply:
Nice catch! I've addressed this - we're now also checking for the generic plans fetching. Let me know if it resolved the issue for you.